discoverers/discoverer_aws_ec2.go

```diff
@@ -127,6 +138,14 @@ func (ec2d *AwsEc2Discoverer) Discover(ctx context.Context) *discovery.Result {
 		return result
 	}
 
+	ssmInstanceStatuses := make(map[string]bool)
+	if ec2d.ssmStatusCheckEnabled {
+		if err := ec2d.collectSsmInstanceStatuses(ctx, ssmInstanceStatuses); err != nil {
```

Comment:
I don't think we should return on error here, we could continue without ssm discovery.
Or maybe just return a warning and continue.